src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/service/NetworkWatchListService.java

```diff
@@ -73,27 +84,74 @@ public void removeWatch(Endpoint endpoint) {
         endpoints.remove(endpoint);
     }
 
+    /**
+     * Handle update slow discovery poll flag request.
+     */
+    public void updateSlowPoll(Endpoint endpoint, boolean isEnabled) {
+        log.info("Watch-list service receive update slow poll flag to {} request for {}", isEnabled, endpoint);
+        endpoints.computeIfPresent(endpoint, (e, endpointContext) ->
+                endpointContext.toBuilder().slowPollEnabled(isEnabled).build());
+    }
+
+    @VisibleForTesting
+    void updateUnhurriedPoll(Endpoint endpoint, boolean isEnabled, long currentTime) {
+        log.info("Watch-list service receive update unhurried poll flag to {} request for {}", isEnabled, endpoint);
+        endpoints.computeIfPresent(endpoint, (e, endpointContext) ->
```

Comment:
Perhaps we should stop doing anything if there is no such endpoint in our `endpoints` map?

Reply:
Yes, we should not. Added additional check.